[Keerthi Vihashini] iP

kvihashini opened the pull request:
# 🌊  dory 

> "just keep swimming" - dory 

dory helps you to remember everything so that you don't have to! 

## 🦭  what's there to sea-anemonemonee 
 - add tasks 
 - mark tasks as done 
 - delete tasks 
 - find tasks
 - list tasks 

## ⚓  where is it? p. sherman 42 wallaby way?
 1. download it from [*here*](https://github.com/kvihashini/ip/releases/tag/A-Jar)
 2. double-click it
 3. have fun!

## 🐟  fish are friends, not food

`showIntro()` is probably the cutest method of the entire code: 


```javascript
 String fish = "                              ....\n"
         + "                             /.... \\\n"
         + "   hi my name is dory    .-`\\ \\   `...')\n"
         + "   and i can help you   ( o   | |      (\n"
         + "    remember things      '-_ / /_./``-._)\n"
         + "                             '\\___\\";
 System.out.println(fish);
```

## ✨  anything left? 

 - [X] adding tasks
 - [X] adding deadlines
 - [X] adding events 
 - [X] deleting everything
 - [ ] pretty GUI

Commit:
Add Gradle support

kvihashini committed:
Add Level 1 - Greet, Echo, Exit

kvihashini committed:
Add Level-2- Add, List

kvihashini committed:
Add Level-3 - Mark as Done

kvihashini committed:
Add Level 4 - ToDos, Events, Deadline

kvihashini committed:
Add A-TextUiTesting - Automated Text UI Testing

kvihashini committed:
Add Level 5 - Handle Errors

kvihashini committed:
Add Level 6 - Delete

kvihashini committed:
Add Level 6 - Delete

kvihashini committed:
Add Level 7 - Save

kvihashini committed:
Add Level 8 - Dates and Times

kvihashini committed:
Merge branch-Level-8

kvihashini committed:
Merge branch-Level-7

kvihashini committed:
Add A-MoreOOP

kvihashini committed:
Add A-Packages

kvihashini committed:
Add A-JUnit

kvihashini committed:
Add A-JavaDoc

kvihashini committed:
Add A-CodingStandard

kvihashini committed:
Add Level-9

kvihashini committed:
Merge branch 'branch-Level-9'

kvihashini committed:
Merge branch 'branch-A-JavaDoc'

kvihashini committed:
Merge branch 'branch-A-CodingStandard'

kvihashini committed:
Fix test bugs

kvihashini committed:
Merge branch 'add-gradle-support' of https://github.com/kvihashini/ip

kvihashini committed:
Add CheckStyle Files

kvihashini committed:
Fix CheckStyle on all files

kvihashini committed:
Merge branch 'branch-A-CheckStyle'

kvihashini committed:
Add A-Gradle

kvihashini committed:
Add Level-10 - GUI

kvihashini committed:
Merge branch 'branch-Level-10'

bharathcs commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,71 @@
+    private Ui ui;
+
+    /**
+     * Constructor for the class.
+     * @param filePath filePath for the data file.
```
Maybe there should be an empty line between description and parameter section? I think thats what they mean [here](https://se-education.org/guides/conventions/java/intermediate.html#comments)

I noticed this in the other javadocs too, so remember to fix those too if this comment makes sense.

```suggestion
     * Constructor for the class.
     *
     * @param filePath filePath for the data file.
```

bharathcs commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    /**
+     * Generates response for user's input.
+     * @param input
```
I'm not sure, but maybe its better for javadocs to include a description for input too, to help ppl immediately grasp the kind of parameters you are looking for? (e.g. input user command typed in console) I'm looking [here](https://se-education.org/guides/conventions/java/intermediate.html#comments)

I noticed this in the other javadocs too, so remember to fix those too if this comment makes sense.

```suggestion
     * Generates response for user's input.
     *
     * @param input user command typed in console
```

bharathcs commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,56 @@
+     */
+    public static Command parse(String fullCommand) {
+        String nextInput = fullCommand.toLowerCase();
+        if (nextInput.contains("delete")) {
+            // extract number from input
+            String taskToBeRemoved = nextInput.replaceAll("[^0-9]", "");
+            // number into integer
+            int numToBeRemoved = Integer.parseInt(taskToBeRemoved) - 1;
+            return new DeleteCommand(numToBeRemoved);
+
+        } else if (nextInput.contains("done")) {
+            // extract number from input
+            String taskToBeMarked = nextInput.replaceAll("[^0-9]", "");
+            // number into integer
+            int numToBeMarked = Integer.parseInt(taskToBeMarked) - 1;
+            return new DoneCommand(numToBeMarked);
+
+        } else if (nextInput.contains("find")) {
+            String inputToMatch = nextInput.replace("find", "").trim();
+            return new FindCommand(inputToMatch);
+
+        } else if (nextInput.contains("bye")) {
+            return new ExitCommand();
+
+        } else if (nextInput.contains("todo") || nextInput.contains("event") || nextInput.contains("deadline")) {
+            return new AddCommand(nextInput);
+
```
Might be neater if you remove the newline before every `else if`?

( Not sure myself, looking at (this)[https://se-education.org/guides/conventions/java/intermediate.html#layout] )

bharathcs commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+
+    private String filePath;
+
+    public Storage(String filePath) {
```
Should we add a description for this constructor (only asking because it has parameters)? I think its possible `filePath` could be explained in javadocs so that it can be used easily.

bharathcs commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,190 @@
+    }
+
+    /**
+     * Displays deleted task.
+     * @param removed Deleted Task.
+     * @param taskList TaskList containing remaining tasks.
```
Really nitpicking here (code too clean haha), but i think these descriptions don't need capitalisations if its not a name, its probably just deleted task right?

bharathcs commented on src/main/java/duke/exceptions/DukeException.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Duke Exception.
+ */
+public class DukeException extends Exception {
```
Since this is a parameter to a public function is this a large scope (can't use short forms)? here
(And for its subclasses)

bharathcs commented on src/main/java/duke/tasktypes/Task.java:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * Returns in a format to save to file.
+     */
+    public String hardDiskSave() {
```
I think this needs to be renamed to match the CS2103t function names (starting with a verb). (and also in the other `.../tasktypes/*` files)

bharathcs reviewed (commented):
LGTM! Its really hard to find errors, good job!

zhenxuantan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,56 @@
+     */
+    public static Command parse(String fullCommand) {
+        String nextInput = fullCommand.toLowerCase();
+        if (nextInput.contains("delete")) {
```
I think that Switch statements would make the code more readable as compared to multiple if else statements. What do you think?

zhenxuantan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+                        String isDone = lineTask[1];
+                        String taskDesc = lineTask[2];
+
+                        boolean doneOrNot = (isDone.equals("true"));
```
Wouldn't this be better ```String isDone = lineTask[1].equals("true");``` instead of delaying the evaluation to doneOrNot?

zhenxuantan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+                            toDo.changeStatus(doneOrNot);
+                            dataForDory.add(toDo);
+                        } else if (taskType.equals("D")) {
+                            String finishBy = lineTask[3];
```
Wouldn't "date" as variable name be better? What do you think?

zhenxuantan commented on src/main/java/duke/tasktypes/Deadlines.java:
```diff
@@ -0,0 +1,50 @@
+/**
+ * Deadline class.
+ */
+public class Deadlines extends Task {
```
I didn't like the fact that Task is a singular noun but the child class extending from it is plural. Do you have any rationale for naming these classes in their plural form?

zhenxuantan commented on src/main/java/duke/tasktypes/Deadlines.java:
```diff
@@ -0,0 +1,50 @@
+        return "D/" + this.getBooleanStatus() + "/" + this.getDescription() + "/" + this.by;
+    }
+
+
```
Is the empty line deliberate?

zhenxuantan reviewed (commented):
Overall, I feel that the code is clean enough, but I think it can be made more readable! 😄

kvihashini committed:
Fix code quality and coding standard violations

kvihashini committed:
Add Shadow plugin to file

Existing JAR file does not contain dependencies

Let’s package the application into an executable fat JAR file

Solves problem by including all the dependencies

kvihashini committed:
Add Assertions

Current files have assumptions that are not verified during runtime.

Let’s add assertions to ensure verification of assumptions.

Identifies bugs in the code that could have violated assumptions.

kvihashini committed:
Improve Code Quality

Current files had issues such as deep nested methods.

Let’s remove arrowhead style code and make happy path prominent.

Improves code quality and long methods.

kvihashini committed:
Merge pull request #1 from kvihashini/branch-A-Assertions

Add Assertions

kvihashini committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Storage.java

kvihashini committed:
Merge pull request #2 from kvihashini/branch-A-CodeQuality

Improve Code Quality

kvihashini committed:
Add Extension - DetectAnomalies

Current files does not check if a clashing event exists.

Let’s add an extension to ensure there is no clash.

Add function to filters through current event dates to check for clash.

kvihashini committed:
Merge branch 'branch-BCD-Extension'

kvihashini committed:
Add A-CI

kvihashini committed:
Improve GUI

kvihashini committed:
Set theme jekyll-theme-minimal

kvihashini committed:
Add User Guide

kvihashini committed:
Set theme jekyll-theme-midnight

kvihashini committed:
Set theme jekyll-theme-minimal

kvihashini committed:
Update README.md

kvihashini committed:
Update README.md

kvihashini committed:
Update User Guide

kvihashini committed:
Fix User Guide bugs

kvihashini committed:
Fix Code Quality

kvihashini committed:
Add Intro Message

kvihashini committed:
Update README.md

kvihashini committed:
Edit CSS

kvihashini committed:
Update README.md
